### PR TITLE
Replacement-PR: Cell face definition pr agchitu

### PR DESCRIPTION
**Pull request by Bellout**
Replaces PR "Cell face definition and bounding boxes" by agchitu.

Commits from that PR have been merged into a new branch,
cell-face-definition-pr-agchitu, branched from latest develop). 
All conflicts were resolved and some formatting changes 
deleted during last PR merge have been added.

### CODE REVIEW

File: FieldOpt/Reservoir/grid/cell.cpp [hunk 1]
```diff
@@ -57,30 +58,76 @@ bool Cell::EnvelopsPoint(Eigen::Vector3d point) {
...
+    // center of the Earth.
+
+    // Here we determine whether the cell is left- or right-handed, which will
+    // later determine whether the vertex indices sets, (0-1-2-3) and (4-5-6-7),
+    // are assigned as "Above" or "Below" cell faces.
+
+    // Make sure there is a good definition of inside and outside of the cell.
+
+    std::array<std::array<int,4>, 6> face_indices_points;
+    if (corners_[0].z() < corners_[4].z()) // z-axis grows towards Earth's center (left-handed)
```
Comment (einar90) [inline FieldOpt/Reservoir/grid/cell.cpp]: Doing this check for every cell is very inefficient. Better to have it once when the grid is initialized.

File: FieldOpt/Reservoir/grid/cell.cpp [hunk 2]
```diff
@@ -57,30 +58,76 @@ bool Cell::EnvelopsPoint(Eigen::Vector3d point) {
...
+    {
+        face_indices_points = _faces_definition_earth_pointing_z;
+    }
+    else // z-axis grows towards the ski - right-handed
+    {
+        face_indices_points = _faces_definition_ski_pointing_z;
```
Comment (einar90) [inline FieldOpt/Reservoir/grid/cell.cpp]: I assume `ski_pointing` should be `sky_pointing`. Also, class member names should end with an underscore.

Comment (Bellout) [inline FieldOpt/Reservoir/grid/cell.cpp]: face_indices_points is only used in this scope, it is not a member of cell... should I rename it?

Comment (Bellout) [inline FieldOpt/Reservoir/grid/cell.cpp]: ahh, you mean _faces_definition_earth_pointing_z? what is the correct convention in this case?

Comment (Bellout) [inline FieldOpt/Reservoir/grid/cell.cpp]: changed it to faces_definition_earth_pointing_z_; also for the other one

**Pull request merged**